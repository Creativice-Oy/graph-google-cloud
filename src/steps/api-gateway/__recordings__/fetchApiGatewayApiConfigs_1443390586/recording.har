{
  "log": {
    "_recordingName": "fetchApiGatewayApiConfigs",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 566,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 566,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:26 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:26.332Z",
        "time": 206,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 206
        }
      },
      {
        "_id": "3045bfccf05203a2c8e9a3c3c0605d72",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.1 gl-node/14.15.4 auth/7.0.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "apigateway.googleapis.com"
            }
          ],
          "headersSize": 561,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://apigateway.googleapis.com/v1/projects/j1-gc-integration-dev-v2/locations/-/apis"
        },
        "response": {
          "bodySize": 553,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 553,
            "text": "{\"apis\":[{\"name\":\"projects/j1-gc-integration-dev-v2/locations/global/apis/example-api-gateway-endpoint\",\"createTime\":\"2021-03-30T21:15:27.820351650Z\",\"updateTime\":\"2021-03-30T21:17:06.954180503Z\",\"displayName\":\"Example API Gateway Endpoint\",\"managedService\":\"example-api-gateway-endpoint-0fo33j5zaqt68.apigateway.j1-gc-integration-dev-v2.cloud.goog\",\"state\":\"ACTIVE\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:27 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:26.550Z",
        "time": 986,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 986
        }
      },
      {
        "_id": "956faced3cbdf6709f1d8e0e952983e6",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.1 gl-node/14.15.4 auth/7.0.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "apigateway.googleapis.com"
            }
          ],
          "headersSize": 608,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://apigateway.googleapis.com/v1/projects/j1-gc-integration-dev-v2/locations/global/apis/example-api-gateway-endpoint:getIamPolicy"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 119,
            "text": "{\"etag\":\"ACAB\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:28 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:27.540Z",
        "time": 869,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 869
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 557,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 557,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:28 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:28.424Z",
        "time": 127,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 127
        }
      },
      {
        "_id": "009f789d5b78c4232d1093ff57657b80",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.1 gl-node/14.15.4 auth/7.0.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "apigateway.googleapis.com"
            }
          ],
          "headersSize": 603,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://apigateway.googleapis.com/v1/projects/j1-gc-integration-dev-v2/locations/global/apis/example-api-gateway-endpoint/configs"
        },
        "response": {
          "bodySize": 645,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 645,
            "text": "{\"apiConfigs\":[{\"name\":\"projects/j1-gc-integration-dev-v2/locations/global/apis/example-api-gateway-endpoint/configs/example-config\",\"createTime\":\"2021-03-30T21:17:10.657320375Z\",\"updateTime\":\"2021-03-31T13:57:35.950462993Z\",\"displayName\":\"Example Config\",\"state\":\"ACTIVE\",\"serviceConfigId\":\"example-config-3u3w8my3mj12x\",\"gatewayServiceAccount\":\"projects/-/serviceAccounts/j1-gc-integration-dev-v2@j1-gc-integration-dev-v2.iam.gserviceaccount.com\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:29 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:28.555Z",
        "time": 861,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 861
        }
      },
      {
        "_id": "37a3386d54b8f84665efd2a185964325",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.1 gl-node/14.15.4 auth/7.0.2"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "apigateway.googleapis.com"
            }
          ],
          "headersSize": 631,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://apigateway.googleapis.com/v1/projects/j1-gc-integration-dev-v2/locations/global/apis/example-api-gateway-endpoint/configs/example-config:getIamPolicy"
        },
        "response": {
          "bodySize": 326,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 326,
            "text": "{\"version\":1,\"etag\":\"BwW+1X9AEg4=\",\"bindings\":[{\"role\":\"roles/apigateway.viewer\",\"members\":[\"user:viragsf@gmail.com\"]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 31 Mar 2021 17:27:30 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-31T17:27:29.421Z",
        "time": 888,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 888
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
