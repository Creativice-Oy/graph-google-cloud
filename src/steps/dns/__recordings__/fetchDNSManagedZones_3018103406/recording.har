{
  "log": {
    "_recordingName": "fetchDNSManagedZones",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 745,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "745"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 558,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 558,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 26 Jan 2021 18:42:02 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-26T18:42:02.768Z",
        "time": 247,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 247
        }
      },
      {
        "_id": "cc91f18c2e474e52b0ef1cbe21ef22cc",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "dns.googleapis.com"
            }
          ],
          "headersSize": 549,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dns.googleapis.com/dns/v1/projects/j1-gc-integration-dev-300716/managedZones"
        },
        "response": {
          "bodySize": 1115,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 1115,
            "text": "{\"managedZones\":[{\"name\":\"example-public-zone\",\"dnsName\":\"example-4d484bdb.com.\",\"description\":\"Example DNS zone\",\"id\":\"6062040873480912950\",\"nameServers\":[\"ns-cloud-e1.googledomains.com.\",\"ns-cloud-e2.googledomains.com.\",\"ns-cloud-e3.googledomains.com.\",\"ns-cloud-e4.googledomains.com.\"],\"creationTime\":\"2021-01-26T18:16:58.899Z\",\"visibility\":\"public\",\"labels\":{\"foo\":\"bar\"},\"kind\":\"dns#managedZone\"},{\"name\":\"example-test-zone\",\"dnsName\":\"myzone.mydomain.com.\",\"description\":\"\",\"id\":\"6171194744693927401\",\"nameServers\":[\"ns-cloud-d1.googledomains.com.\",\"ns-cloud-d2.googledomains.com.\",\"ns-cloud-d3.googledomains.com.\",\"ns-cloud-d4.googledomains.com.\"],\"creationTime\":\"2021-01-26T18:08:56.992Z\",\"visibility\":\"public\",\"kind\":\"dns#managedZone\"},{\"name\":\"example-zone\",\"dnsName\":\"crawleroo.com.\",\"description\":\"\",\"id\":\"778386319953398157\",\"nameServers\":[\"ns-cloud-a1.googledomains.com.\",\"ns-cloud-a2.googledomains.com.\",\"ns-cloud-a3.googledomains.com.\",\"ns-cloud-a4.googledomains.com.\"],\"creationTime\":\"2021-01-22T16:48:27.272Z\",\"dnssecConfig\":{\"state\":\"on\",\"defaultKeySpecs\":[{\"keyType\":\"keySigning\",\"algorithm\":\"rsasha256\",\"keyLength\":2048,\"kind\":\"dns#dnsKeySpec\"},{\"keyType\":\"zoneSigning\",\"algorithm\":\"rsasha256\",\"keyLength\":1024,\"kind\":\"dns#dnsKeySpec\"}],\"nonExistence\":\"nsec3\",\"kind\":\"dns#managedZoneDnsSecConfig\"},\"visibility\":\"public\",\"kind\":\"dns#managedZone\"}],\"kind\":\"dns#managedZonesListResponse\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 26 Jan 2021 18:42:03 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-26T18:42:03.028Z",
        "time": 447,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 447
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
