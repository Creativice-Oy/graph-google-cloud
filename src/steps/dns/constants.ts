export const STEP_DNS_MANAGED_ZONES = 'fetch-dns-managed-zones';

export const DNS_MANAGED_ZONE_ENTITY_CLASS = 'DomainZone';
export const DNS_MANAGED_ZONE_ENTITY_TYPE = 'google_dns_managed_zone';
