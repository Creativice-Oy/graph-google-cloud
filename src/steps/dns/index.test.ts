import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
import { fetchDNSManagedZones } from '.';
import { integrationConfig } from '../../../test/config';
import { Recording, setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { DNS_MANAGED_ZONE_ENTITY_TYPE } from './constants';

describe('#fetchDNSManagedZones', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchDNSManagedZones',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchDNSManagedZones(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === DNS_MANAGED_ZONE_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DomainZone'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_dns_managed_zone' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          displayName: { type: 'string' },
          domainName: { type: 'string' },
          visibility: { type: 'string' },
          kind: { type: 'string' },
          dnssec: { type: 'string' },
          keySigningAlgorithm: { type: 'string' },
          zoneSigningAlgorithm: { type: 'string' },
          createdOn: { type: 'number' },
        },
      },
    });
  });
});
