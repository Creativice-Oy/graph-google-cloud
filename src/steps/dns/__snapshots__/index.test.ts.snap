// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchDNSManagedZones should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "DomainZone",
      ],
      "_key": "5302175939166426816",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "creationTime": "2021-02-18T00:04:21.953Z",
            "description": "Example DNS zone",
            "dnsName": "example-981611f4.com.",
            "id": "5302175939166426816",
            "kind": "dns#managedZone",
            "name": "example-public-zone",
            "nameServers": Array [
              "ns-cloud-c1.googledomains.com.",
              "ns-cloud-c2.googledomains.com.",
              "ns-cloud-c3.googledomains.com.",
              "ns-cloud-c4.googledomains.com.",
            ],
            "visibility": "public",
          },
        },
      ],
      "_type": "google_dns_managed_zone",
      "createdOn": 1613606661953,
      "description": "Example DNS zone",
      "displayName": "example-public-zone",
      "dnssec": undefined,
      "domainName": "example-981611f4.com.",
      "id": "5302175939166426816",
      "keySigningAlgorithm": undefined,
      "kind": "dns#managedZone",
      "name": "example-public-zone",
      "visibility": "public",
      "zoneSigningAlgorithm": undefined,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_dns_managed_zone",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
